src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,156 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int INITIAL_LIST_SIZE = 10;
+    private static final int FIRST_INDEX = 1;
+    private int size;
+    private T[] elementData;
+    private Node<T> first;
+    private Node<T> last;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    MyLinkedList() {
+        first = new Node<T>(null, null, null);
+        last = new Node<T>(null, null,null);
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<T>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+            first.prev = null;
+        }
+        last.next = newNode;
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> currentNode = findNode(index);
+        Node<T> newNode = new Node<T>(currentNode.prev, value, currentNode);
+        if (index == 0) {
+            first = newNode;
+            newNode.next.prev = newNode;
+            size++;
```

Comment:
size++ should be only at the and of the method

Reply:
Leaved one size++ at the end of method